app/api/api_v1/routers/search.py

```diff
@@ -132,18 +136,35 @@ def search_documents(
     the search database. The continuation token can be used to get the next set of
     results from the search database. See the request schema for more details.
     """
+    try:
+        allowed_corpora_ids = decode_config_token(app_token, PUBLIC_APP_URL)
+    except PyJWTError as e:
+        _LOGGER.error(e)
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Could not decode configuration token",
```

Comment:
I'm not sure if this is leaking too much information - as we move towards the commercial aspects - just a note, happy to keep it like this for now until there is need for change.